invisible_cities/cities/components.py

```diff
@@ -357,40 +357,33 @@ def mcsensors_from_file(paths     : List[str],
                  Run number for database
     """
 
-    pmt_ids  = load_db.DataPMT (db_file, run_number).SensorID
+    pmt_ids  = load_db.DataPMT(db_file, run_number).SensorID
 
     for file_name in paths:
         sns_resp = mcinfo_io.load_mcsensor_response_df(file_name              ,
```

Comment:
I think we should skip empty or corrupted files with a `try except`, producing a warning that the file couldn't be processed?

Reply:
The errors that you suggest are already caught in `dst_io.load_dst` which is used in `load_mcsensor_response_df`. I'd need to make the warnings catchable.

Reply:
> I think we should skip empty or corrupted files with a try except, producing a warning that the file couldn't be processed?

Skipping the files with a warning may not be a good idea. As far as we can tell from these last 6 months of productions, corrupted files do not appear to be a common issue (in fact, we've seen 0 corrupted files), while completely empty files are usually a symptom of errors in the production (errors in the job configuration of nexus, for example) and a good reason to stop the processing. Empty events may be legit, and should be filtered out (i.e. skipped but counted).

Reply:
I think that making a job crash because one file is empty or corrupt is annoying. 
I agree that we should not simply forget about those files, but that could be dealt with at the nexus stage, looking at nexus outputs and making sure that we understand why some files are not ok. What I mean is that trying to process them doesn't seem necessarily the right place for this sanity check to me. 
However, both approaches can be done, I have no strong opinion on any of them.

Reply:
So, to conclude:
- No protection against corrupted files, can be added afterwards if really needed 
- A filter that will filter events that left no signal in the detector (the filter in the flow, not just skipping the event in the source)

Reply:
If there's no signal left in the sensors for an event the event doesn't appear in the table in the input file and, as such, can't be skipped or processed because it's not there.

I also can't remove the protection for corrupt files without catchable warnings since the underlying reader `dst_io.load_dst` already converts the exception to a warning. If this is to be changed it should be a separate PR with all the checks required, I think.

What should I do? Revert to having the `MCEventNotFound` error for an empty?

This is getting a bit confused since most of the limitations related to this error vs. warning discussion come from functions that aren't directly related to the PR.